### PR TITLE
new(tests) EIP-7069 RETURNDATALOAD and RETURNDATACOPY

### PR DESCRIPTION
**Pull request by shemnon**
## 🗒️ Description

Test new RETURNDATACOPY semantics across both EOF and Legacy.
Test RETURNDATALOAD semantics. Includes boundary conditions on both.
## ✅ Checklist
- [ ] All: Set appropriate labels for the changes.
- [ ] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Comment by shemnon** [comment]
Will rebase and force push once #522 is merged.

**Review by marioevz — COMMENTED** [review]
Some comments, thanks!

File: tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        TestAddress: Account(balance=10**18, nonce=tx.nonce),
+        address_entry_point: Account(
+            nonce=1,
+            code=Op.NOOP
+            # First, create a "dirty" area, so we can check zero overwrite
+            + Op.PUSH1(1)
+            + Op.PUSH0
+            + Op.SUB
+            + Op.MSTORE(0x00, Op.DUP1)
+            + Op.MSTORE(0x20, Op.DUP1)
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py:96]: Yes it would, that's what took most of the time planning for this PR was contemplating how I would prove the zero extension at interesting lengths.  We could also make a test or two of absurd lengths too (2k/128k)

File: tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            + Op.POP
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py:97]: Yea, I was considering wether to make the python longer or make the bytecode smaller.  I'll golf in other repos.

File: tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        + Op.SSTORE(slot_code_worked, value_code_worked)
+        + Op.RETURN(0, size)
+    )
+    match container_style:
+        case "EOF":
+            pre[address_caller] = Account(
+                code=Container(
+                    sections=[
+                        Section.Code(
+                            code=Op.EXTCALL(address_returner, 0, 0, 0) + code_under_test,
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py]: Done.

File: tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    slot_result_start = 0x1000
+
+    pre = {
+        TestAddress: Account(balance=10**18, nonce=tx.nonce),
+        address_entry_point: Account(
+            nonce=1,
+            code=Container(
+                sections=[
+                    Section.Code(
+                        code=Op.EXTDELEGATECALL(address_returner, 0, 0)
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py]: Done.

File: tests/prague/eip7692_eof_v1/eip7069_extcall/test_address_space_extension.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+                    Section.Code(
+                        code=Op.SSTORE(
+                            slot_target_call_status,
+                            target_opcode(Op.CALLDATALOAD(0), *call_suffix),
+                        )
+                        + Op.RETURNDATACOPY(0, 0, Op.RETURNDATASIZE)
+                        + Op.SSTORE(slot_target_returndata, Op.MLOAD(0))
+                        + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip7069_extcall/test_address_space_extension.py]: Done.

File: tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (marioevz) [reply tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py]: Sounds good, I created an issue, so I'll ping you once that is done and we can revisit these tests then.

**Review by marioevz — APPROVED** [review]
Looks great, loved the latest changes it's pretty clean, thanks for this.
